server/bleep/src/indexes/file.rs

```diff
@@ -153,6 +156,64 @@ impl File {
     }
 }
 
+fn should_index<P: AsRef<Path>>(p: &P) -> bool {
+    let path = p.as_ref();
+
+    const EXT_BLACKLIST: &[&str] = &["png", "jpg", "jpeg", "ico", "ttf", "otf", "woff2"];
+
+    let Some(ext) = path.extension() else {
+        return true;
+    };
+
+    let ext = ext.to_string_lossy();
+    if EXT_BLACKLIST.contains(&&*ext) {
+        return false;
+    }
+
+    static VENDOR_PATTERNS: Lazy<HashMap<&'static str, SmallVec<[Regex; 1]>>> = Lazy::new(|| {
+        let patterns: &[(&[&str], &[&str])] = &[
+            (
+                &["go", "proto"],
+                &["^(vendor|third_party)/.*\\.\\w+$", "\\w+\\.pb\\.go$"],
+            ),
+            (
+                &["js", "jsx", "ts", "tsx", "css", "md", "json", "txt", "conf"],
+                &["^(node_modules|vendor|dist)/.*\\.\\w+$"],
+            ),
+        ];
+
+        patterns
+            .iter()
+            .flat_map(|(exts, rxs)| exts.iter().map(move |&e| (e, rxs)))
+            .map(|(ext, rxs)| {
+                let regexes = rxs
+                    .iter()
+                    .filter_map(|source| match Regex::new(source) {
+                        Ok(r) => Some(r),
+                        Err(e) => {
+                            warn!(%e, "failed to compile vendor regex {source:?}");
+                            None
+                        }
+                    })
+                    .collect();
+
+                (ext, regexes)
+            })
+            .collect()
+    });
+
+    match VENDOR_PATTERNS.get(&*ext) {
+        None => true,
+        Some(rxs) => !rxs.iter().any(|r| r.is_match(&path.to_string_lossy())),
+    }
+}
+
+// Empirically calculated using:
```

Comment:
What's the corpus you have done this on? 30C/line seems a bit low?

Reply:
Tested using the `bleep` source code here, surprised me as well